conan/tools/files/patches.py

```diff
@@ -118,3 +118,27 @@ def apply_conandata_patches(conanfile):
         else:
             raise ConanException("The 'conandata.yml' file needs a 'patch_file' or 'patch_string'"
                                  " entry for every patch to be applied")
+
+
+def export_conandata_patches(conanfile):
+    """
+    Exports patches stored in 'conanfile.conan_data' (read from 'conandata.yml' file). It will export
+    all the patches under 'patches' entry that matches the given 'conanfile.version'. If versions are
+    not defined in 'conandata.yml' it will export all the patches directly under 'patches' keyword.
+    """
+
+    patches = conanfile.conan_data.get('patches')
+
+    if isinstance(patches, dict):
+        assert conanfile.version, "Can only be exported if conanfile.version is already defined"
+        entries = patches.get(conanfile.version, [])
+    elif isinstance(patches, Iterable):
+        entries = patches
+    else:
+        return
```

Comment:
No silent return if it is a string for example. If it is None when not defined, return earlier.

Reply:
> No silent return if it is a string for example.

What do you mean by that ? I know patch can be string then not exported then.

> If it is None when not defined, return earlier.

Why the behaviour should be different from `apply_conandata_patches` ?

Reply:
Oh, I wasn't aware that the ``apply_conandata_patches`` had this behavior. Then it can be left the same in this PR, and I will create an issue to check this.